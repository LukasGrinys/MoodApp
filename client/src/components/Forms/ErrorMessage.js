import React from 'react';
import styles from './Forms.module.scss';

const ErrorMessage = ({children}) => {
    return (
        <div className={styles.errorMessage}>
            {children}
        </div>
    );
};

export default ErrorMessage;