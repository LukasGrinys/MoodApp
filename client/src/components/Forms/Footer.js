import React from 'react';
import styles from './Forms.module.scss'

const Footer = ({children}) => {
    return (
        <div className={styles.formFooter}>
            {children}
        </div>
    );
};

export default Footer;