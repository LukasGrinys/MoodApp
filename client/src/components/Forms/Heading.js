import React from 'react';
import styles from './Forms.module.scss';

const Heading = ({children}) => {
    return <h1 className={styles.heading}>{children}</h1>;
};

export default Heading;