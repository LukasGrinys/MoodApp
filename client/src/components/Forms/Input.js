import React from 'react';
import styles from './Forms.module.scss';

const Input = ({type, name, value, controlEvents, placeholder}) => {
    const { handleChange, handleBlur } = controlEvents;

    if (type === 'text' || type === 'email' || type === 'password') {
        return (
            <input 
                type={type} 
                name={name} 
                value={value}
                className={styles.textInput}
                onChange={handleChange}
                onBlur={handleBlur}
            />
        );
    }

    if (type === 'textarea') {
        return (
            <textarea 
                name={name}
                onChange={handleChange}
                onBlur={handleBlur}
                value={value}
                placeholder={placeholder}
                className={styles.textArea}
            />
        )
    }

};

Input.defaultProps = {
    type : 'text',
    name : '',
    value : '',
    controlEvents : {
        handleBlur : () => {},
        handleChange : () => {}
    },
    placeholder: ''
}

export default Input;