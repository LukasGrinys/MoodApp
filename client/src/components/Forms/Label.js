import React from 'react';
import styles from './Forms.module.scss';

const Label = ({children}) => {
    return <span className={styles.label}>{children}</span>
};

export default Label;