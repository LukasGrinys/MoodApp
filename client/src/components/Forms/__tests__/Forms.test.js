import React from 'react';

import Footer from '../Footer';
import ErrorMessage from '../ErrorMessage';
import Heading from '../Heading';
import Input from '../Input';
import Label from '../Label';

import { shallow } from 'enzyme';

describe('Input field', () => {
    const handleChangeMock = jest.fn();
    const handleBlurMock = jest.fn();

    const inputProps = {
        type: 'text',
        name: 'Test',
        value: '',
        controlEvents: {
            handleChange: handleChangeMock,
            handleBlur: handleBlurMock
        }, 
        placeholder: 'testPlaceholder'
    }
    const inputWrapper = shallow(<Input {...inputProps}/>);

    test('Renders propely', () => {
        expect(inputWrapper).toMatchSnapshot();
    });

    test('Input element has proper props', () => {
        expect(inputWrapper.find('input').props()).toEqual({
            className: "textInput",
            name: "Test",
            onBlur: handleBlurMock,
            onChange: handleChangeMock,
            type: "text",
            value: ""
        })
    });

    test('Calls handleChange when typed', () => {
        inputWrapper.find('input').simulate('change', {
            target: {
                name: inputProps.name,
                value: ''
            }
        });
        inputWrapper.update();

        expect(handleChangeMock).toHaveBeenCalledTimes(1);
    });

    test('Calls handleBlur when blurred', () => {
        inputWrapper.find('input').simulate('blur');
        inputWrapper.update();

        expect(handleBlurMock).toHaveBeenCalledTimes(1);
    });

    test('Renders textarea if "textarea" is passed as type prop', () => {
        const textareaProps = {
            type: 'textarea'
        }
        const textareaWrapper = shallow(<Input {...textareaProps}/>);

        expect(textareaWrapper.find('textarea').length).toBe(1);
    });
});

describe('Static elements renders properly', () => {
    const text = 'test';
    const footerWrapper = shallow(<Footer>{text}</Footer>)

    test('Footer renders properly', () => {
        expect(footerWrapper).toMatchSnapshot();
    });

    test('Footer renders text passed as children', () => {
        expect(footerWrapper.text()).toBe(text);
    });

    const labelWrapper = shallow(<Label>{text}</Label>)

    test('Label renders properly', () => {
        expect(labelWrapper).toMatchSnapshot();
    });

    test('Label renders text passed as children', () => {
        expect(labelWrapper.text()).toBe(text);
    });

    const headingWrapper = shallow(<Heading>{text}</Heading>)

    test('Heading renders properly', () => {
        expect(headingWrapper).toMatchSnapshot();
    });

    test('Heading renders text passed as children', () => {
        expect(headingWrapper.text()).toBe(text);
    });

    const errorWrapper = shallow(<ErrorMessage>{text}</ErrorMessage>)

    test('Error renders properly', () => {
        expect(errorWrapper).toMatchSnapshot();
    });

    test('Error renders text passed as children', () => {
        expect(errorWrapper.text()).toBe(text);
    });
})

