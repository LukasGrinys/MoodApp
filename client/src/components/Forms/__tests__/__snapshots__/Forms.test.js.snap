// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field Renders propely 1`] = `
<input
  className="textInput"
  name="Test"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  type="text"
  value=""
/>
`;

exports[`Static elements renders properly Error renders properly 1`] = `
<div
  className="errorMessage"
>
  test
</div>
`;

exports[`Static elements renders properly Footer renders properly 1`] = `
<div
  className="formFooter"
>
  test
</div>
`;

exports[`Static elements renders properly Heading renders properly 1`] = `
<h1
  className="heading"
>
  test
</h1>
`;

exports[`Static elements renders properly Label renders properly 1`] = `
<span
  className="label"
>
  test
</span>
`;
